### DISCUSSION TITLE
How to configure pool light?

### DISCUSSION THREAD

**Discussion by jkmhb** [tagyoureit/nodejs-poolController#606] [App and O/S Config Questions]
Hi. I have sequent 8 relay board. I have it 50% configured but was wondering how to configure for pool light so I can turn it on and off? Thanks.

**Comment by rstrouse** [comment]
Which pool light do you have?  Is it a color changing light?

You must add a circuit under the gears menu/AUX Circuits tab and assign the type of circuit from the dropdown.  If this is simply a generic light then choose Light.   Next in the connection dropdown choose your REM server and then select the relay that you have the light hooked to.  Finally, press the Save Circuit button and you are off to the races.   This will now activate the light whenever you press the button on the dashPanel.

**Reply by jkmhb** [reply]
It is not color changing. A regular pool light. I can now see it using your advice, but it does not trip the relay when I turn it on, even though I defined the relay. I can manually turn on relay using 8relind command

**Comment by rstrouse** [comment]
Please take a screenshot of the circuit configuration.

EDIT:  Did you assign the relay to the circuit by selecting the REM server and the relay that is wired to the light?
![image](https://user-images.githubusercontent.com/47839015/190011760-a4c24cd5-c8f9-4864-ac4a-b3a31b2431ab.png)

**Comment by jkmhb** [comment]
![image](https://user-images.githubusercontent.com/50624053/190021483-95d21818-f78c-45d7-9ce6-758f78b74f08.png)![image](https://user-images.githubusercontent.com/50624053/190021633-5f80e263-b1e5-401c-816d-d331667c49d0.png)
![image](https://user-images.githubusercontent.com/50624053/190021720-a276a3a7-5404-4966-ab3d-3b032458e2f0.png)

**Reply by rstrouse** [reply]
Hmmm... where did your pool circuit go?  Did you delete it?

**Reply by jkmhb** [reply]
I must have deleted it. I put it back, I used "no connection". it still does not work.

**Comment by rstrouse** [comment]
If you shut down and re-start njsPC it will put the proper pool circuit back.  It looks like we need to remove the delete circuit button for pool and/or spa circuits related to the body since these circuits are tied to the body and have a very special function.

Please capture a replay while pressing the feature button on dashPanel.  To do this follow the capture replay instructions from this wiki.  You do not need to capture the configuration reload as I am only looking for log messages possibly displayed when the request to trigger the relay is made.

https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel

**Comment by jkmhb** [comment]
I deleted my "pool" restarted the pi and 'pool" resppeared.Then I captured the logs (attached). I have a hubitat and I see a lot of [ ](error: undefined). Thank you for all your help!
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/9568673/replay.zip)

**Reply by rstrouse** [reply]
Go into the Controller ==> Interfaces tab and disable the Hubitat interface for the moment.  I see the request being processed successfully from REM.  Also, do you have any feeds or triggers set on the relay?

Also, it appears you do not have any RS485 devices so you can go into the Controller ==> RS485 Tab and uncheck the Enabled checkbox for the primary port then press save.

Do the steps above and lets see if the relay latches when you turn on the light.

**Reply by jkmhb** [reply]
I deleted hubitat restarted the pi. Also  unchecked rs485. It still did not work. I did not have any triggers or feeds associated with this relay. Maybe I should delete everything for this relay and define it the way we know works, in the correct order? Again,it does work with the 8relind command

**Reply by rstrouse** [reply]
Now that you have done this shoot another replay.  The order should not matter.  Does it work by clicking on the relay in REM?

**Reply by rstrouse** [reply]
Wait a minute.  I just noticed the address you have set up for the Sequent 8 Relay.  When you use the 8relind command what is the stack level you are using?  Run `i2cdetect -y 1` from the command line and post those results here.  45 (0x2d) is outside the bounds of the Sequent Relay Board addresses.  Normally if you are using stack level 0 then it should be 39 (0x27) and one should be added to the address until you have up to 8 boards.

**Reply by jkmhb** [reply]
I get 24, 2d, and 4a. My board is defined as 3 for the relay board. I also have a sequent smart fan and home automation boards.smart fan is 1 and home automation is 2.

**Reply by rstrouse** [reply]
That stack level jumper does not work that way.  Each of the boards have their own address space so jumpering is not needed unless you have two of the same board.  Honestly, in an attempt to simplify the the I2c addresses, Sequent has made it all more difficult.  Sadly, the document the I2c addresses for the relay at 0x20 to 0x27 but even that is not correct.  Based on the code if you are at stack level 3 then the address should be 46 (0x2E).  When you run the commands are you doing `8relind 2 read` or are you doing `8relind 3 read`

Does the light come on when you click on the button in REM?

**Reply by jkmhb** [reply]
Yes, now it does. I moved my definition for the relay card to 36 or x24.when I run 8relind -list I get 1 board(s) detected 
Id:3 I am doing 8relind 3 read get 0 when off

**Reply by rstrouse** [reply]
Now reassign the relay to the circuit and it should work.

**Reply by jkmhb** [reply]
I reassigned the circuit  and it does not work. It is close

**Reply by rstrouse** [reply]
Hmmm enable all 8 relays.  Then turn on the light from the dashPanel and let’s see if they remapped the relays.

if you are running pm2 you can monitor the log for REM to check for errors.  Run pm2 monit.

**Comment by jkmhb** [comment]
I activated all  relays. I get no change in pool controller dashpanel. I can turn on/off all 8 relays in relayequipmentmanager. I see nothing in pm2, except a lot of TS033: Could not write. These are not timestamped. See attached![image](https://user-images.githubusercontent.com/50624053/190410420-d3803fbd-dd6c-48f1-9bab-15ca38f76953.png)

**Reply by jkmhb** [reply]
It is all working now. My relay board was not active in REM. I found out by rerunning capture as outlined earlier. Thank you both for all your help!

**Reply by rstrouse** [reply]
Awesome!  What was the final address for your Relay 8 in REM.  36 (0x24)?

**Reply by jkmhb** [reply]
Yes. 0x24.
